### PR TITLE
Servicify menu, close menu on admin for medium screens

### PR DESCRIPTION
**Pull request by rigelk**

### CODE REVIEW

File: client/src/app/core/routing/menu-guard.service.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  }
+
+  static close () {
+    MenuGuards.guards.push(CloseMenuGuard)
+    return CloseMenuGuard
+  }
+}
+
+abstract class MenuGuard implements CanActivate, CanDeactivate<any> {
+  public display = true
```
Comment (Chocobozzz) [inline client/src/app/core/routing/menu-guard.service.ts]: Remove `public`

File: client/src/app/core/routing/menu-guard.service.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  }
+  canActivate (): boolean {
+    // small screens already have the site-wide onResize from screenService
+    // > medium screens have enough space to fit the administrative menus
+    if (!this.screen.isInMobileView() && this.screen.isInMediumView()) {
+      this.menu.isMenuDisplayed = this.display
+    }
+    return true
+  }
+  public canDeactivate = this.canActivate
```
Comment (Chocobozzz) [inline client/src/app/core/routing/menu-guard.service.ts]: Remove `public` and put attributes below `display = true`

File: client/src/app/core/routing/menu-guard.service.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    MenuGuards.guards.push(CloseMenuGuard)
+    return CloseMenuGuard
+  }
+}
+
+abstract class MenuGuard implements CanActivate, CanDeactivate<any> {
+  public display = true
+  constructor (protected menu: MenuService, protected screen: ScreenService, display: boolean) {
+    this.display = display
+  }
```
Comment (Chocobozzz) [inline client/src/app/core/routing/menu-guard.service.ts]: Add ` \n`

File: client/src/app/core/routing/menu-guard.service.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  canActivate (): boolean {
+    // small screens already have the site-wide onResize from screenService
+    // > medium screens have enough space to fit the administrative menus
+    if (!this.screen.isInMobileView() && this.screen.isInMediumView()) {
+      this.menu.isMenuDisplayed = this.display
+    }
+    return true
+  }
```
Comment (Chocobozzz) [inline client/src/app/core/routing/menu-guard.service.ts]: Add ` \n`

File: client/src/app/core/routing/menu-guard.service.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+import { CanActivate, CanDeactivate } from '@angular/router'
+import { MenuService } from '@app/core/menu'
+import { ScreenService } from '@app/shared/misc/screen.service'
+
+@Injectable()
+export class MenuGuards {
+  public static guards: any = []
+
+  static open () {
+    MenuGuards.guards.push(OpenMenuGuard)
```
Comment (Chocobozzz) [inline client/src/app/core/routing/menu-guard.service.ts]: Why do you need to push `OpenMenuGuard`?

Comment (rigelk) [reply client/src/app/core/routing/menu-guard.service.ts]: To make the providers in the module definition properly consider them.

File: client/src/app/core/routing/menu-guard.service.ts (hunk 2 above)
Comment (rigelk) [reply client/src/app/core/routing/menu-guard.service.ts]: This is a function, so it should be placed below the function it copies.

Comment (Chocobozzz) [reply client/src/app/core/routing/menu-guard.service.ts]: No: https://travis-ci.org/Chocobozzz/PeerTube/jobs/659521986

Comment (rigelk) [reply client/src/app/core/routing/menu-guard.service.ts]: That's news to me !

File: client/src/app/core/routing/menu-guard.service.ts (hunk 5 above)
Comment (Chocobozzz) [reply client/src/app/core/routing/menu-guard.service.ts]: Okay then please use a `Set` or check duplicates so you don't add multiple times the same guard

File: client/src/app/core/routing/menu-guard.service.ts [hunk 6]
```diff
@@ -0,0 +1,47 @@
+import { Injectable } from '@angular/core'
+import { CanActivate, CanDeactivate } from '@angular/router'
+import { MenuService } from '@app/core/menu'
+import { ScreenService } from '@app/shared/misc/screen.service'
+
+@Injectable()
+export class MenuGuards {
+  public static guards: any = []
```
Comment (Chocobozzz) [inline client/src/app/core/routing/menu-guard.service.ts]: I think you can type this to `MenuGuard[]`

**Comment by rigelk** [comment]
@Chocobozzz I figured the current use is simple enough to just list both cases explicitly for now.

**Pull request merged**